src/textual/widgets/_tabbed_content.py

```diff
@@ -430,13 +432,19 @@ def add_pane(
         pane = self._set_id(pane, tabs.tab_count + 1)
         assert pane.id is not None
         pane.display = False
+
+        async def _add_part(awaitable: AwaitComplete | AwaitMount) -> None:
+            await awaitable
+
         return AwaitComplete(
-            tabs.add_tab(
-                ContentTab(pane._title, pane.id),
-                before=before if before is None else ContentTab.add_prefix(before),
-                after=after if after is None else ContentTab.add_prefix(after),
+            _add_part(
+                tabs.add_tab(
+                    ContentTab(pane._title, pane.id),
+                    before=before if before is None else ContentTab.add_prefix(before),
+                    after=after if after is None else ContentTab.add_prefix(after),
+                )
             ),
-            self.get_child_by_type(ContentSwitcher).mount(pane),
+            _add_part(self.get_child_by_type(ContentSwitcher).mount(pane)),
```

Comment:
This actually seems like a flaw. The `AwaitCompete` class runs its awaitables concurrently. This looks like something that should run in serial.